materialize-snowflake/snowflake.go

```diff
@@ -547,38 +550,65 @@ func (d *transactor) Load(it *pm.LoadIterator, loaded func(int, json.RawMessage)
 	if len(subqueries) == 0 {
 		return nil // Nothing to load.
 	}
-	var loadAllSQL = strings.Join(subqueries, "\nUNION ALL\n") + ";"
 
-	// Issue a join of the target table and (now staged) load keys,
-	// and send results to the |loaded| callback.
-	rows, err := d.load.conn.QueryContext(sf.WithStreamDownloader(ctx), loadAllSQL)
-	if err != nil {
-		return fmt.Errorf("querying Load documents: %w", err)
-	}
-	defer rows.Close()
+	group, groupCtx := errgroup.WithContext(ctx)
+	group.SetLimit(MaxConcurrentLoads)
+	var mutex sync.Mutex
 
-	for rows.Next() {
-		var binding int
-		var document stdsql.RawBytes
+	for iLoop, queryLoop := range subqueries {
+		var query = queryLoop
+		var i = iLoop
+		group.Go(func() error {
+			var b = d.bindings[i]
+
+			log.WithField("table", b.target.Identifier).Info("load: starting querying documents")
+			// Issue a join of the target table and (now staged) load keys,
+			// and send results to the |loaded| callback.
+			rows, err := d.load.conn.QueryContext(sf.WithStreamDownloader(groupCtx), query)
```

Comment:
I don't think using the same connection for all goroutines is going to result in a throughput benefit. A single connection must execute queries sequentially, so sharing this same connection will still end up with the load queries still being run one after another. I do think a `*sql.Conn` is thread safe (although I'm not 100% sure), but it [has locking](https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/database/sql/sql.go;l=504-524;drc=894d24d617bb72d6e1bed7b143f9f7a0ac16b844) that will result in concurrent calls waiting for each other to complete.

Using a separate connection for each load goroutine seems like a viable option though. It would probably be fine to just do `db.QueryContext` and let the `db` manage connection pooling, rather than keeping a reference to one specific connection for the life of the transactor.